### PR TITLE
Bump dependencies

### LINKED ISSUE #98: Bump dependencies
It would be great if `torchsde` could be installed into an environment with `numpy==1.20.*` and `scipy==1.6.*`
https://github.com/google-research/torchsde/blob/4c84649176f0d041e6fa9604d307b743246a4237/setup.py#L37

### PR DESCRIPTION
**Pull request by dhirschfeld**
Resolves #98

### CODE REVIEW

**Comment by dhirschfeld** [comment]
Figured I'd see if it breaks anything! 😅

**Comment by dhirschfeld** [comment]
ping! Anyone able to take a look? At least kick off the CI?

File: setup.py [hunk 1]
```diff
@@ -34,7 +34,7 @@
     description="SDE solvers and stochastic adjoint sensitivity analysis in PyTorch.",
     url="https://github.com/google-research/torchsde",
     packages=setuptools.find_packages(exclude=['benchmarks', 'diagnostics', 'examples', 'tests']),
-    install_requires=['torch>=1.6.0', 'numpy==1.19.*', 'boltons>=20.2.1', 'trampoline>=0.1.2', 'scipy==1.5.*'],
+    install_requires=['torch>=1.6.0', 'numpy==1.20.*', 'boltons>=20.2.1', 'trampoline>=0.1.2', 'scipy==1.6.*'],
```
Comment (lxuechen) [inline setup.py:37]: Could we change this to >=1.19.0 or something like that? Thanks.

Comment (lxuechen) [reply setup.py]: Also 'scipy==1.5.*' => `'scipy>=1.5.*'`?

Comment (dhirschfeld) [reply setup.py]: Done!

**Review by lxuechen — APPROVED** [review]

**Comment by lxuechen** [comment]
The Py3.6 runs failed due to scipy not being compatible. I just replicated this behaviour on my machine as well. I think a simple workaround is to enforce the version more strictly using the Python version fetch during run time in `setup.py`. WDYT? 

Happy to also accept other potentially better sol'ns. Thanks for the patience again.

**Comment by patrick-kidger** [comment]
@lxuechen One possible resolution -- I'd be happy to bump the minimum Python version to 3.7 if you are? I [hear on the grapevine](https://twitter.com/ezyang/status/1379955134894071810) that that's the minimum version PyTorch is going to be supporting pretty soon anyway.

**Comment by dhirschfeld** [comment]
> *I'd be happy to bump the minimum Python version to 3.7 if you are?*

That works for me. The latest `scipy` already has a minimum version of py37:
https://github.com/scipy/scipy/pull/13081

**Comment by lxuechen** [comment]
> > _I'd be happy to bump the minimum Python version to 3.7 if you are?_
> 
> That works for me. The latest `scipy` already has a minimum version of py37:
> [scipy/scipy#13081](https://github.com/scipy/scipy/pull/13081)

For now, I think since torch hasn't dropped support for Python3.5, I'd prefer that we don't up the min version yet. Feel free to leave a TODO in `setup.py` so that we may revisit in the future.

**Comment by lxuechen** [comment]
Thanks again for sending in this PR, @dhirschfeld ! I still plan to merge this even though there's been a huge delay. If you could do a rebase from master and add the runtime Python version checks, I can merge this right away. 

Again, sorry for the unresponsiveness on my end.

**Comment by dhirschfeld** [comment]
Hi @lxuechen, sorry I've been meaning to circle back to this but I'm pretty smashed with the end of the fin-year coming up. I might not get around to revisiting for a couple of weeks so I'm happy for you to take this over or to start fresh if you's like it in sooner - whatever is easiest.

**Comment by Sayam753** [comment]
Hi @lxuechen . Is this PR ready to be merged? or are there some TODOs still remaining?

**Comment by Sayam753** [comment]
Here is the PR https://github.com/Sayam753/torchsde/pull/1 at my fork which bumps the dependencies. I can roll out a PR in this repo if this is the right way to update dependencies :)

**Comment by dhirschfeld** [comment]
The last `scipy` compatible with `py36` was 1.5.4:
https://github.com/scipy/scipy/blob/v1.5.4/setup.py#L539

Since `scipy=1.7.0` isn't compatible with `py36` `pip` should never have tried installing it. Unfortunately `pip`  isn't smart enough to work that out :/

**Comment by dhirschfeld** [comment]
@lxuechen / @patrick-kidger can you please approve the CI so that I can test out a solution using [platform specific dependencies](https://setuptools.readthedocs.io/en/latest/userguide/dependency_management.html#platform-specific-dependencies)?

image

**Comment by patrick-kidger** [comment]
Approved the CI but I don't see any platform-specific dependencies in the current PR.
(Sorry that this has been such a faff to resolve, by the way.)

**Comment by dhirschfeld** [comment]
> I don't see any platform-specific dependencies in the current PR.

Right, sorry - didn't push. Looks like it needs to be approved on every commit? That's not a great UX/DX for anyone! 😬

**Comment by dhirschfeld** [comment]
Same issue with `numpy` - [`1.19.5`](https://github.com/numpy/numpy/blob/v1.19.5/setup.py#L468) was the last version to support `py36`

**Comment by dhirschfeld** [comment]
> Sorry that this has been such a faff to resolve, by the way.

No worries - looks like we got there in the end! :tada:

**Comment by patrick-kidger** [comment]
LGTM! Thanks for contributing, and resolving this issue.

**Comment by Sayam753** [comment]
Thanks for the fast response and merging in the PR :)

**Comment by dhirschfeld** [comment]
Just wanted to check if a release was imminent? If not I might patch the deps on the existing `conda` package so that it's installable with the latest verions.

**Comment by patrick-kidger** [comment]
Installing from GitHub is the main option we support at the moment (we do need to put it on PyPI/conda already), so probably no imminent release.

**Comment by dhirschfeld** [comment]
`conda-forge` *usually* prefer to create packages from `pypi` sdists however it's also possible to create packages from GitHub releases.

**Pull request merged**